# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/02/06 22:21:22-08:00 matthew@wil.cx 
#   [IPV4]: ipconfig should use memmove() instead of strcpy()
#   
#   strcpy is undefined if src and dest overlap.  That's clearly possible
#   here with a sufficiently deep path on the server.  Use memmove instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
# 
# net/ipv4/ipconfig.c
#   2005/02/06 22:21:02-08:00 matthew@wil.cx +1 -1
#   `### Change the comments to ChangeSet|1.1547 below
#   [IPV4]: ipconfig should use memmove() instead of strcpy()
#   
#   strcpy is undefined if src and dest overlap.  That's clearly possible
#   here with a sufficiently deep path on the server.  Use memmove instead.
#   
#   Signed-off-by: Matthew Wilcox <matthew@wil.cx>
#   Signed-off-by: David S. Miller <davem@davemloft.net>
#   
#   [IPV4]: ipconfig should use memmove() instead of strcpy()
# 
diff -Nru a/net/ipv4/ipconfig.c b/net/ipv4/ipconfig.c
--- a/net/ipv4/ipconfig.c	2005-02-09 09:12:44 -08:00
+++ b/net/ipv4/ipconfig.c	2005-02-09 09:12:44 -08:00
@@ -1162,7 +1162,7 @@
 		if (*cp == ':')
 			*cp++ = '\0';
 		addr = in_aton(name);
-		strcpy(name, cp);
+		memmove(name, cp, strlen(cp) + 1);
 	} else
 		addr = INADDR_NONE;
 
